src/main/java/uk/gov/hmcts/ccd/domain/service/common/StateReferenceService.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.ccd.domain.service.common;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.domain.casestate.EnablingConditionParser;
+import uk.gov.hmcts.ccd.domain.model.definition.EventPostStateDefinition;
+
+@Service
+public class StateReferenceService {
+
+    private final EnablingConditionParser enablingConditionParser;
+
+    @Inject
+    public StateReferenceService(EnablingConditionParser enablingConditionParser) {
+        this.enablingConditionParser = enablingConditionParser;
+    }
+
+    public Optional<String> evaluatePostStateCondition(List<EventPostStateDefinition> eventPostStateDefinitions,
+                                                       Map<String, JsonNode> caseEventData) {
```

Comment:
should get the whole case data in input since enabling conditions can use any case field

Reply:
for now only on case event fields defined, not on any case field